#!/usr/bin/env python

PACKAGE = "robot_localizer"

from dynamic_reconfigure.parameter_generator_catkin import *
import math

gen = ParameterGenerator()
#       Name                           Type   Level Description                 Default    Min        Max
gen.add("n_particles",                 int_t, 0, "n_particles",                  300,     0,        500)
gen.add("initial_uncertainty_xy",      double_t, 0, "initial_uncertainty_xy",         1,     0,          5)
gen.add("initial_uncertainty_theta",   double_t, 0, "initial_uncertainty_theta",    0.5,     0,  2*math.pi)
gen.add("variance_scale",              double_t, 0, "variance_scale",              0.15,     0,          1)
gen.add("n_particles_average",         int_t, 0, "n_particles_average",           20,     1,        200)
gen.add("linear_var_thresh",           double_t, 0, "linear_var_thresh",           0.05,     0,        0.5)
gen.add("angular_var_thresh",          double_t, 0, "angular_var_thresh",           0.2,     0,    math.pi)
gen.add("min_valid_pts",               int_t, 0, "min_valid_pts",                 10,     0,        360)
gen.add("d_thresh",                    double_t, 0, "d_thresh",                     0.2,  0.01,          1)
gen.add("a_thresh",                    double_t, 0, "a_thresh",               math.pi/6,  0.01,  math.pi/2)
gen.add("laser_max_distance",          double_t, 0, "laser_max_distance",           2.0,   0.5,        5.0)


exit(gen.generate(PACKAGE, "robot_localizer", "ParticleFilter"))
